<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Running fedup with multiple test sets • fedup</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Running fedup with multiple test sets">
<meta property="og:description" content="fedup">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">fedup</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>

  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/fedup_doubleTest.html">Running fedup with two test sets</a>
    </li>
    <li>
      <a href="../articles/fedup_mutliTest.html">Running fedup with multiple test sets</a>
    </li>
    <li>
      <a href="../articles/fedup_singleTest.html">Running fedup with a single test set</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/rosscm/fedup/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><script src="fedup_mutliTest_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Running fedup with multiple test sets</h1>
                        <h4 class="author">Catherine Ross</h4>


      <small class="dont-index">Source: <a href="https://github.com/rosscm/fedup/blob/master/vignettes/fedup_mutliTest.rmd"><code>vignettes/fedup_mutliTest.rmd</code></a></small>
      <div class="hidden name"><code>fedup_mutliTest.rmd</code></div>

    </div>



<p>This is an R package that tests for enrichment and depletion of user-defined pathways using a Fisher’s exact test. The method is designed for versatile pathway annotation formats (eg. gmt, txt, xlsx) to allow the user to run pathway analysis on custom annotations. This package is also integrated with Cytoscape to provide network-based pathway visualization that enhances the interpretability of the results.</p>
<p>This vignette will explain how to use <code>fedup</code> when testing multiple sets of genes for pathway enrichment and depletion.</p>
<div id="system-prerequisites" class="section level1">
<h1 class="hasAnchor">
<a href="#system-prerequisites" class="anchor"></a>System prerequisites</h1>
<p><strong>R version</strong> ≥ 4.1<br><strong>R packages</strong>:</p>
<ul>
<li>
<strong>CRAN</strong>: openxlsx, tibble, dplyr, data.table, ggplot2, ggthemes, forcats, RColorBrewer</li>
<li>
<strong>Bioconductor</strong>: RCy3</li>
</ul>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Install <code>fedup</code> from Bioconductor:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"fedup"</span><span class="op">)</span></code></pre></div>
<p>Or install the development version from Github:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"rosscm/fedup"</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Load necessary packages:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rosscm/fedup">fedup</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="running-the-package" class="section level1">
<h1 class="hasAnchor">
<a href="#running-the-package" class="anchor"></a>Running the package</h1>
<div id="input-data" class="section level2">
<h2 class="hasAnchor">
<a href="#input-data" class="anchor"></a>Input data</h2>
<p>Load test genes (<code>geneMulti</code>) and pathways annotations (<code>pathwaysGMT</code>):</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneMulti</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">pathwaysGMT</span><span class="op">)</span></code></pre></div>
<p>Take a look at the data structure:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">geneMulti</span><span class="op">)</span>
<span class="co">#&gt; List of 13</span>
<span class="co">#&gt;  $ background       : chr [1:17804] "SLCO4A1" "PGRMC2" "LDLR" "RABL3" ...</span>
<span class="co">#&gt;  $ FASN_negative    : chr [1:379] "SLCO4A1" "PGRMC2" "LDLR" "RABL3" ...</span>
<span class="co">#&gt;  $ FASN_positive    : chr [1:298] "CDC34" "PRKCE" "SMARCC2" "EIF3A" ...</span>
<span class="co">#&gt;  $ ACACA_negative   : chr [1:276] "SLCO4A1" "PGRMC2" "RABL3" "C12orf49" ...</span>
<span class="co">#&gt;  $ ACACA_positive   : chr [1:313] "RAB18" "MORC2" "DPM1" "BCL2" ...</span>
<span class="co">#&gt;  $ LDLR_negative    : chr [1:394] "LCMT1" "SREBF2" "DOLPP1" "ALG9" ...</span>
<span class="co">#&gt;  $ LDLR_positive    : chr [1:386] "RAB18" "PHF12" "SMARCD1" "USP14" ...</span>
<span class="co">#&gt;  $ SREBF1_negative  : chr [1:262] "SREBF2" "MBTPS2" "CNOT2" "TSC1" ...</span>
<span class="co">#&gt;  $ SREBF1_positive  : chr [1:218] "PHF5A" "RNF7" "KHDRBS1" "UBR4" ...</span>
<span class="co">#&gt;  $ SREBF2_negative  : chr [1:299] "LDLR" "C12orf49" "PAPSS1" "ARL1" ...</span>
<span class="co">#&gt;  $ SREBF2_positive  : chr [1:315] "KCTD5" "MTF2" "BRD9" "TSC1" ...</span>
<span class="co">#&gt;  $ C12orf49_negative: chr [1:376] "LDLR" "ACSL1" "SREBF2" "MOGS" ...</span>
<span class="co">#&gt;  $ C12orf49_positive: chr [1:215] "SAR1A" "BAG6" "UBA5" "TMED10" ...</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">pathwaysGMT</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; List of 6</span>
<span class="co">#&gt;  $ REGULATION OF PLK1 ACTIVITY AT G2 M TRANSITION%REACTOME%R-HSA-2565942.1          : chr [1:84] "CSNK1E" "DYNLL1" "TUBG1" "CKAP5" ...</span>
<span class="co">#&gt;  $ GLYCEROPHOSPHOLIPID BIOSYNTHESIS%REACTOME%R-HSA-1483206.4                        : chr [1:126] "PCYT1B" "PCYT1A" "PLA2G4D" "PLA2G4B" ...</span>
<span class="co">#&gt;  $ MITOTIC PROPHASE%REACTOME DATABASE ID RELEASE 74%68875                           : chr [1:134] "SETD8" "NUMA1" "NCAPG2" "LMNB1" ...</span>
<span class="co">#&gt;  $ ACTIVATION OF NF-KAPPAB IN B CELLS%REACTOME%R-HSA-1169091.1                      : chr [1:67] "PSMA6" "PSMA3" "PSMA4" "PSMA1" ...</span>
<span class="co">#&gt;  $ CD28 DEPENDENT PI3K AKT SIGNALING%REACTOME DATABASE ID RELEASE 74%389357         : chr [1:22] "CD28" "THEM4" "AKT1" "TRIB3" ...</span>
<span class="co">#&gt;  $ UBIQUITIN-DEPENDENT DEGRADATION OF CYCLIN D%REACTOME DATABASE ID RELEASE 74%75815: chr [1:52] "PSMA6" "PSMA3" "PSMA4" "PSMA1" ...</span></code></pre></div>
<p>To see more info on this data, run <code><a href="../reference/geneDouble.html">?geneDouble</a></code> or <code><a href="../reference/pathwaysGMT.html">?pathwaysGMT</a></code>. You could also run <code><a href="https://rdrr.io/r/utils/example.html">example("prepInput", package = "fedup")</a></code> or <code><a href="https://rdrr.io/r/utils/example.html">example("readPathways", package = "fedup")</a></code> to see exactly how the data was generated using the <code><a href="../reference/prepInput.html">prepInput()</a></code> and <code><a href="../reference/readPathways.html">readPathways()</a></code> functions. <code><a href="https://rdrr.io/r/utils/Question.html">?</a></code> and <code><a href="https://rdrr.io/r/utils/example.html">example()</a></code> can be used on any other functions mentioned here to see their documentation and run examples.</p>
<p>The sample <code>geneMulti</code> list object contains thirteen vector elements: <code>background</code>, <code>FASN_negative</code>, <code>FASN_positive</code>, <code>ACACA_negative</code>, <code>ACACA_positive</code>, <code>LDLR_negative</code>, <code>LDLR_positive</code>, <code>SREBF1_negative</code>, <code>SREBF1_positive</code>, <code>SREBF2_negative</code>, <code>SREBF2_positive</code>, <code>C12orf49_negative</code>, and <code>C12orf49_positive</code>. The <code>background</code> consists of all genes that the test sets (in this case all sets besides <code>background</code>) will be compared against. <code>FASN_negative</code> consists of genes that form <strong>negative genetic interactions</strong> with the FASN gene after CRISPR-Cas9 knockout. <code>FASN_positive</code> consists of genes that form <strong>positve genetic interactions</strong> with <em>FASN</em>. The remaining elements contain genes forming genetic interactions with their respective genes (<em>ACACA</em>, <em>LDLR</em>, <em>SREBF1</em>, <em>SREBF2</em>, <em>C12orf49</em>). If you’re interested in seeing how this data set was constructed, check out the <a href="https://github.com/rosscm/fedup/blob/main/inst/script/genes.R">code</a>. Also, the paper the data was taken from is found <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7566881/">here</a>.</p>
<p>As an example, <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=FASN"><em>FASN</em></a> is a fatty acid synthase, so we would expect to see <strong>enrichment</strong> of the negative interactions for pathways associated with <em>sensitization</em> of fatty acid synthesis, as well as <strong>enrichment</strong> of the positive interactions for pathways associated with <em>suppression</em> of the function. Conversely, we expect to find <strong>depletion</strong> for pathways not at all involved with <em>FASN</em> biology. Let’s see!</p>
</div>
<div id="pathway-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#pathway-analysis" class="anchor"></a>Pathway analysis</h2>
<p>Now use <code>runFedup</code> on the sample data:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fedupRes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runFedup.html">runFedup</a></span><span class="op">(</span><span class="va">geneMulti</span>, <span class="va">pathwaysGMT</span><span class="op">)</span>
<span class="co">#&gt; Warning in inputObject(genes, pathways): 28.081% of genes overlap across your test gene sets (enrichment</span>
<span class="co">#&gt;         results may be similar across tests).</span>
<span class="co">#&gt; Running fedup with:</span>
<span class="co">#&gt;  =&gt; 12 test set(s)</span>
<span class="co">#&gt;   + FASN_negative: 379 genes</span>
<span class="co">#&gt;   + FASN_positive: 298 genes</span>
<span class="co">#&gt;   + ACACA_negative: 276 genes</span>
<span class="co">#&gt;   + ACACA_positive: 313 genes</span>
<span class="co">#&gt;   + LDLR_negative: 394 genes</span>
<span class="co">#&gt;   + LDLR_positive: 386 genes</span>
<span class="co">#&gt;   + SREBF1_negative: 262 genes</span>
<span class="co">#&gt;   + SREBF1_positive: 218 genes</span>
<span class="co">#&gt;   + SREBF2_negative: 299 genes</span>
<span class="co">#&gt;   + SREBF2_positive: 315 genes</span>
<span class="co">#&gt;   + C12orf49_negative: 376 genes</span>
<span class="co">#&gt;   + C12orf49_positive: 215 genes</span>
<span class="co">#&gt;  =&gt; 17804 background genes</span>
<span class="co">#&gt;  =&gt; 1437 pathway annotations</span>
<span class="co">#&gt; All done!</span></code></pre></div>
<p>The <code>fedupRes</code> output is a list of length <code><a href="https://rdrr.io/r/base/length.html">length(which(names(geneMulti) != "background"))</a></code>, corresponding to the number of test sets in <code>geneMulti</code> (i.e., 12).</p>
<p>View <code>fedup</code> results for <code>FASN_negative</code> sorted by pvalue:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">set</span> <span class="op">&lt;-</span> <span class="st">"FASN_negative"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">fedupRes</span><span class="op">[[</span><span class="va">set</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">fedupRes</span><span class="op">[[</span><span class="va">set</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">status</span> <span class="op">==</span> <span class="st">"enriched"</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;                                                                                                                                          pathway</span>
<span class="co">#&gt; 1:                                                                      ASPARAGINE N-LINKED GLYCOSYLATION%REACTOME DATABASE ID RELEASE 74%446203</span>
<span class="co">#&gt; 2: BIOSYNTHESIS OF THE N-GLYCAN PRECURSOR (DOLICHOL LIPID-LINKED OLIGOSACCHARIDE, LLO) AND TRANSFER TO A NASCENT PROTEIN%REACTOME%R-HSA-446193.1</span>
<span class="co">#&gt; 3:                                                  DISEASES ASSOCIATED WITH N-GLYCOSYLATION OF PROTEINS%REACTOME DATABASE ID RELEASE 74%3781860</span>
<span class="co">#&gt; 4:                                                        INTRA-GOLGI AND RETROGRADE GOLGI-TO-ER TRAFFIC%REACTOME DATABASE ID RELEASE 74%6811442</span>
<span class="co">#&gt; 5:                                                                         RAB REGULATION OF TRAFFICKING%REACTOME DATABASE ID RELEASE 74%9007101</span>
<span class="co">#&gt; 6:                                                                                            DISEASES OF GLYCOSYLATION%REACTOME%R-HSA-3781865.1</span>
<span class="co">#&gt;    size real_frac expected_frac fold_enrichment   status</span>
<span class="co">#&gt; 1:  286  8.179420    1.53336329        5.334300 enriched</span>
<span class="co">#&gt; 2:   78  3.693931    0.42125365        8.768901 enriched</span>
<span class="co">#&gt; 3:   17  2.110818    0.09548416       22.106472 enriched</span>
<span class="co">#&gt; 4:  183  4.749340    0.99415862        4.777246 enriched</span>
<span class="co">#&gt; 5:  120  3.693931    0.62345540        5.924933 enriched</span>
<span class="co">#&gt; 6:  139  3.957784    0.74702314        5.298074 enriched</span>
<span class="co">#&gt;                                     real_gene       pvalue       qvalue</span>
<span class="co">#&gt; 1:      MOGS,DOLPP1,ALG9,ALG12,ALG3,MPDU1,... 1.596605e-12 2.294321e-09</span>
<span class="co">#&gt; 2:      DOLPP1,ALG9,ALG12,ALG3,MPDU1,ALG8,... 6.358461e-09 4.568554e-06</span>
<span class="co">#&gt; 3:       MOGS,ALG9,ALG12,ALG3,MPDU1,MGAT2,... 3.054616e-08 1.463161e-05</span>
<span class="co">#&gt; 4:  ARL1,RAB18,RAB3GAP2,VPS52,NAA35,TMED9,... 2.516179e-07 9.039372e-05</span>
<span class="co">#&gt; 5: RAB18,TSC1,RAB3GAP2,TSC2,TBC1D20,RAB10,... 4.945154e-07 1.421237e-04</span>
<span class="co">#&gt; 6:       MOGS,ALG9,ALG12,ALG3,MPDU1,MGAT2,... 7.240716e-07 1.734151e-04</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">fedupRes</span><span class="op">[[</span><span class="va">set</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">fedupRes</span><span class="op">[[</span><span class="va">set</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">status</span> <span class="op">==</span> <span class="st">"depleted"</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;                                                               pathway size</span>
<span class="co">#&gt; 1:                        GPCR LIGAND BINDING%REACTOME%R-HSA-500792.3  454</span>
<span class="co">#&gt; 2: OLFACTORY SIGNALING PATHWAY%REACTOME DATABASE ID RELEASE 74%381753  396</span>
<span class="co">#&gt; 3:       CLASS A 1 (RHODOPSIN-LIKE RECEPTORS)%REACTOME%R-HSA-373076.7  323</span>
<span class="co">#&gt; 4:             NEURONAL SYSTEM%REACTOME DATABASE ID RELEASE 74%112316  379</span>
<span class="co">#&gt; 5:           PEPTIDE LIGAND-BINDING RECEPTORS%REACTOME%R-HSA-375276.5  195</span>
<span class="co">#&gt; 6:             KERATINIZATION%REACTOME DATABASE ID RELEASE 74%6805567  217</span>
<span class="co">#&gt;    real_frac expected_frac fold_enrichment   status    real_gene      pvalue</span>
<span class="co">#&gt; 1: 0.0000000     2.3702539       0.0000000 depleted              0.000318537</span>
<span class="co">#&gt; 2: 0.0000000     1.9096832       0.0000000 depleted              0.001508862</span>
<span class="co">#&gt; 3: 0.0000000     1.6906313       0.0000000 depleted              0.003316944</span>
<span class="co">#&gt; 4: 0.5277045     2.0950348       0.2518834 depleted KCNK2,PRKAB1 0.026904721</span>
<span class="co">#&gt; 5: 0.0000000     1.0166255       0.0000000 depleted              0.057057149</span>
<span class="co">#&gt; 6: 0.0000000     0.8425073       0.0000000 depleted              0.079543380</span>
<span class="co">#&gt;        qvalue</span>
<span class="co">#&gt; 1: 0.01760530</span>
<span class="co">#&gt; 2: 0.05420587</span>
<span class="co">#&gt; 3: 0.10361845</span>
<span class="co">#&gt; 4: 0.42024004</span>
<span class="co">#&gt; 5: 0.57670567</span>
<span class="co">#&gt; 6: 0.67813171</span></code></pre></div>
<p>Here we see the strongest enrichment for the <code>ASPARAGINE N-LINKED GLYCOSYLATION</code> pathway. Given that <em>FASN</em> mutant cells show a strong dependence on lipid uptake, this enrichment for negative interactions with genes involved in glycosylation is expected. We also see significant enrichment for other related pathways, including <code>DISEASES ASSOCIATED WITH N-GLYCOSYLATION OF PROTEINS</code> and <code>DISEASES OF GLYCOSYLATION</code>. Conversely, we see significant depletion for functions not associated with these processes, such as <code>OLFACTORY SIGNALING PATHWAY</code>, <code>GPCR LIGAND BINDING</code> and <code>KERATINIZATION</code>. Nice!</p>
<p>Let’s also view <code>fedup</code> results for <code>FASN_positive</code>, sorted by pvalue:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">set</span> <span class="op">&lt;-</span> <span class="st">"FASN_positive"</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">fedupRes</span><span class="op">[[</span><span class="va">set</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">fedupRes</span><span class="op">[[</span><span class="va">set</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">status</span> <span class="op">==</span> <span class="st">"enriched"</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;                                                                                          pathway</span>
<span class="co">#&gt; 1:     L13A-MEDIATED TRANSLATIONAL SILENCING OF CERULOPLASMIN EXPRESSION%REACTOME%R-HSA-156827.3</span>
<span class="co">#&gt; 2: GTP HYDROLYSIS AND JOINING OF THE 60S RIBOSOMAL SUBUNIT%REACTOME DATABASE ID RELEASE 74%72706</span>
<span class="co">#&gt; 3:                    CAP-DEPENDENT TRANSLATION INITIATION%REACTOME DATABASE ID RELEASE 74%72737</span>
<span class="co">#&gt; 4:                                      EUKARYOTIC TRANSLATION INITIATION%REACTOME%R-HSA-72613.3</span>
<span class="co">#&gt; 5:                               TRANSLATION INITIATION COMPLEX FORMATION%REACTOME%R-HSA-72649.3</span>
<span class="co">#&gt; 6:          RIBOSOMAL SCANNING AND START CODON RECOGNITION%REACTOME DATABASE ID RELEASE 74%72702</span>
<span class="co">#&gt;    size real_frac expected_frac fold_enrichment   status</span>
<span class="co">#&gt; 1:  112  7.382550     0.4718041        15.64749 enriched</span>
<span class="co">#&gt; 2:  113  7.382550     0.4774208        15.46340 enriched</span>
<span class="co">#&gt; 3:  120  7.382550     0.5167378        14.28684 enriched</span>
<span class="co">#&gt; 4:  120  7.382550     0.5167378        14.28684 enriched</span>
<span class="co">#&gt; 5:   59  5.369128     0.2583689        20.78086 enriched</span>
<span class="co">#&gt; 6:   59  5.369128     0.2583689        20.78086 enriched</span>
<span class="co">#&gt;                                 real_gene       pvalue       qvalue</span>
<span class="co">#&gt; 1: EIF3A,RPL35,EIF3D,RPS3,EIF3G,EIF4H,... 9.628857e-18 8.562503e-15</span>
<span class="co">#&gt; 2: EIF3A,RPL35,EIF3D,RPS3,EIF3G,EIF4H,... 1.191719e-17 8.562503e-15</span>
<span class="co">#&gt; 3: EIF3A,RPL35,EIF3D,RPS3,EIF3G,EIF4H,... 4.970934e-17 1.785808e-14</span>
<span class="co">#&gt; 4: EIF3A,RPL35,EIF3D,RPS3,EIF3G,EIF4H,... 4.970934e-17 1.785808e-14</span>
<span class="co">#&gt; 5:  EIF3A,EIF3D,RPS3,EIF3G,EIF4H,RPS5,... 5.796507e-15 1.388264e-12</span>
<span class="co">#&gt; 6:  EIF3A,EIF3D,RPS3,EIF3G,EIF4H,RPS5,... 5.796507e-15 1.388264e-12</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">fedupRes</span><span class="op">[[</span><span class="va">set</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">fedupRes</span><span class="op">[[</span><span class="va">set</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">status</span> <span class="op">==</span> <span class="st">"depleted"</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;                                                                         pathway</span>
<span class="co">#&gt; 1:                                  GPCR LIGAND BINDING%REACTOME%R-HSA-500792.3</span>
<span class="co">#&gt; 2:                       NEURONAL SYSTEM%REACTOME DATABASE ID RELEASE 74%112316</span>
<span class="co">#&gt; 3:           OLFACTORY SIGNALING PATHWAY%REACTOME DATABASE ID RELEASE 74%381753</span>
<span class="co">#&gt; 4:                 CLASS A 1 (RHODOPSIN-LIKE RECEPTORS)%REACTOME%R-HSA-373076.7</span>
<span class="co">#&gt; 5:                        G ALPHA (I) SIGNALLING EVENTS%REACTOME%R-HSA-418594.6</span>
<span class="co">#&gt; 6: TRANSMISSION ACROSS CHEMICAL SYNAPSES%REACTOME DATABASE ID RELEASE 74%112315</span>
<span class="co">#&gt;    size real_frac expected_frac fold_enrichment   status real_gene      pvalue</span>
<span class="co">#&gt; 1:  454 0.0000000      2.370254       0.0000000 depleted           0.002390509</span>
<span class="co">#&gt; 2:  379 0.0000000      2.095035       0.0000000 depleted           0.005261657</span>
<span class="co">#&gt; 3:  396 0.0000000      1.909683       0.0000000 depleted           0.007449873</span>
<span class="co">#&gt; 4:  323 0.0000000      1.690631       0.0000000 depleted           0.017309826</span>
<span class="co">#&gt; 5:  396 0.3355705      2.106268       0.1593199 depleted    AHCYL1 0.034808044</span>
<span class="co">#&gt; 6:  238 0.0000000      1.314311       0.0000000 depleted           0.035700272</span>
<span class="co">#&gt;        qvalue</span>
<span class="co">#&gt; 1: 0.03240718</span>
<span class="co">#&gt; 2: 0.05953545</span>
<span class="co">#&gt; 3: 0.07989155</span>
<span class="co">#&gt; 4: 0.13667154</span>
<span class="co">#&gt; 5: 0.21016453</span>
<span class="co">#&gt; 6: 0.21198880</span></code></pre></div>
<p>Results for any test set can be indexed by its name:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">fedupRes</span><span class="op">)</span>
<span class="co">#&gt;  [1] "FASN_negative"     "FASN_positive"     "ACACA_negative"   </span>
<span class="co">#&gt;  [4] "ACACA_positive"    "LDLR_negative"     "LDLR_positive"    </span>
<span class="co">#&gt;  [7] "SREBF1_negative"   "SREBF1_positive"   "SREBF2_negative"  </span>
<span class="co">#&gt; [10] "SREBF2_positive"   "C12orf49_negative" "C12orf49_positive"</span></code></pre></div>
</div>
<div id="dot-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#dot-plot" class="anchor"></a>Dot plot</h2>
<p>Prepare data for plotting via <code>dplyr</code> and <code>tidyr</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fedupPlot</span> <span class="op">&lt;-</span> <span class="va">fedupRes</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span>.id <span class="op">=</span> <span class="st">"set"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html">separate</a></span><span class="op">(</span>col <span class="op">=</span> <span class="st">"set"</span>, into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"set"</span>, <span class="st">"sign"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">qvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>log10qvalue <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">qvalue</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>pathway <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"\\%.*"</span>, <span class="st">""</span>, <span class="va">pathway</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Since we’re dealing with multiple test sets here, it’s important we create the <code>set</code> and <code>sign</code> columns in <code>fedupPlot</code> to distinguish between them. Take a look at <code><a href="https://dplyr.tidyverse.org/reference/bind.html">?dplyr::bind_rows</a></code> for details on how the output <code>fedup</code> results list (<code>fedupRes</code>) was bound into a single dataframe and <code><a href="https://tidyr.tidyverse.org/reference/separate.html">?tidyr::separate</a></code> for how the <code>set</code> and <code>sign</code> columns were created.</p>
<p>Plot significant results (qvalue &lt; 0.05) in the form of a dot plot via <code>plotDotPlot</code>. Facet the points by the <code>set</code> and <code>sign</code> columns and colour by <code>sign</code>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotDotPlot.html">plotDotPlot</a></span><span class="op">(</span>
        df <span class="op">=</span> <span class="va">fedupPlot</span>,
        xVar <span class="op">=</span> <span class="st">"log10qvalue"</span>,
        yVar <span class="op">=</span> <span class="st">"pathway"</span>,
        xLab <span class="op">=</span> <span class="st">"-log10(qvalue)"</span>,
        fillVar <span class="op">=</span> <span class="st">"sign"</span>,
        fillLab <span class="op">=</span> <span class="st">"Genetic interaction"</span>,
        fillCol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#6D90CA"</span>, <span class="st">"#F6EB13"</span><span class="op">)</span>,
        sizeVar <span class="op">=</span> <span class="st">"fold_enrichment"</span>,
        sizeLab <span class="op">=</span> <span class="st">"Fold enrichment"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span><span class="op">(</span><span class="st">"sign ~ set"</span>, scales <span class="op">=</span> <span class="st">"free_y"</span>, space <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>strip.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></code></pre></div>
<p><img src="figures/fedupDotPlot-1.png" width="100%"></p>
<p>Look at all those chick… enrichments! This is a bit overwhelming, isn’t it? How do we interpret these 244 fairly redundant pathways in a way that doesn’t hurt our tired brains even more? Oh I know, let’s first try another ggplot-based plot.</p>
<p>We can instead plot the degree of fold enrichment for a subset of pathways across our test sets. First, select the top 20 results to plot from <code>fedupRes</code>:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">topPath</span> <span class="op">&lt;-</span> <span class="va">fedupRes</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span>.id <span class="op">=</span> <span class="st">"set"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">fold_enrichment</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>View the selected pathways:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">topPath</span><span class="op">)</span>
<span class="co">#&gt;  [1] "DISEASES ASSOCIATED WITH N-GLYCOSYLATION OF PROTEINS%REACTOME DATABASE ID RELEASE 74%3781860"                                 </span>
<span class="co">#&gt;  [2] "TRANSLATION INITIATION COMPLEX FORMATION%REACTOME%R-HSA-72649.3"                                                              </span>
<span class="co">#&gt;  [3] "RIBOSOMAL SCANNING AND START CODON RECOGNITION%REACTOME DATABASE ID RELEASE 74%72702"                                         </span>
<span class="co">#&gt;  [4] "ACTIVATION OF THE MRNA UPON BINDING OF THE CAP-BINDING COMPLEX AND EIFS, AND SUBSEQUENT BINDING TO 43S%REACTOME%R-HSA-72662.3"</span>
<span class="co">#&gt;  [5] "MTORC1-MEDIATED SIGNALLING%REACTOME%R-HSA-166208.3"                                                                           </span>
<span class="co">#&gt;  [6] "COHESIN LOADING ONTO CHROMATIN%REACTOME%R-HSA-2470946.1"                                                                      </span>
<span class="co">#&gt;  [7] "FORMATION OF THE TERNARY COMPLEX, AND SUBSEQUENTLY, THE 43S COMPLEX%REACTOME%R-HSA-72695.2"                                   </span>
<span class="co">#&gt;  [8] "CELL DIVISION%REACTOME%R-HSA-68884.3"                                                                                         </span>
<span class="co">#&gt;  [9] "SIGNALING BY PDGFRA EXTRACELLULAR DOMAIN MUTANTS%REACTOME DATABASE ID RELEASE 74%9673770"                                     </span>
<span class="co">#&gt; [10] "SIGNALING BY PDGFRA TRANSMEMBRANE, JUXTAMEMBRANE AND KINASE DOMAIN MUTANTS%REACTOME DATABASE ID RELEASE 74%9673767"           </span>
<span class="co">#&gt; [11] "MRNA DECAY BY 5' TO 3' EXORIBONUCLEASE%REACTOME DATABASE ID RELEASE 74%430039"                                                </span>
<span class="co">#&gt; [12] "ACTIVATION OF PPARGC1A (PGC-1ALPHA) BY PHOSPHORYLATION%REACTOME%R-HSA-2151209.1"                                              </span>
<span class="co">#&gt; [13] "CD22 MEDIATED BCR REGULATION%REACTOME DATABASE ID RELEASE 74%5690714"                                                         </span>
<span class="co">#&gt; [14] "ATF4 ACTIVATES GENES IN RESPONSE TO ENDOPLASMIC RETICULUM STRESS%REACTOME DATABASE ID RELEASE 74%380994"                      </span>
<span class="co">#&gt; [15] "FORMATION OF SENESCENCE-ASSOCIATED HETEROCHROMATIN FOCI (SAHF)%REACTOME DATABASE ID RELEASE 74%2559584"                       </span>
<span class="co">#&gt; [16] "L13A-MEDIATED TRANSLATIONAL SILENCING OF CERULOPLASMIN EXPRESSION%REACTOME%R-HSA-156827.3"                                    </span>
<span class="co">#&gt; [17] "GTP HYDROLYSIS AND JOINING OF THE 60S RIBOSOMAL SUBUNIT%REACTOME DATABASE ID RELEASE 74%72706"                                </span>
<span class="co">#&gt; [18] "CHOLESTEROL BIOSYNTHESIS%REACTOME%R-HSA-191273.4"                                                                             </span>
<span class="co">#&gt; [19] "ROLE OF LAT2 NTAL LAB ON CALCIUM MOBILIZATION%REACTOME%R-HSA-2730905.2"                                                       </span>
<span class="co">#&gt; [20] "CONDENSATION OF PROMETAPHASE CHROMOSOMES%REACTOME%R-HSA-2514853.2"</span></code></pre></div>
<p>Now subset <code>fedupRes</code> across all test sets for the pathways stored in <code>topPath</code>:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fedupPlot</span> <span class="op">&lt;-</span> <span class="va">fedupRes</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span>.id <span class="op">=</span> <span class="st">"set"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html">separate</a></span><span class="op">(</span>col <span class="op">=</span> <span class="st">"set"</span>, into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"set"</span>, <span class="st">"sign"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">pathway</span> <span class="op">%in%</span> <span class="va">topPath</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>pathway <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"\\%.*"</span>, <span class="st">""</span>, <span class="va">pathway</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>sign <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">status</span> <span class="op">==</span> <span class="st">"depleted"</span>, <span class="st">"none"</span>, <span class="va">sign</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>sign <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">sign</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"negative"</span>, <span class="st">"positive"</span>, <span class="st">"none"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">set</span>, <span class="va">pathway</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html">top_n</a></span><span class="op">(</span><span class="fl">1</span>, wt <span class="op">=</span> <span class="va">fold_enrichment</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Plot via <code>plotDotPlot</code>, this time using <code>set</code> as our x-axis variable:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotDotPlot.html">plotDotPlot</a></span><span class="op">(</span>
        df <span class="op">=</span> <span class="va">fedupPlot</span>,
        xVar <span class="op">=</span> <span class="st">"set"</span>,
        yVar <span class="op">=</span> <span class="st">"pathway"</span>,
        xLab <span class="op">=</span> <span class="cn">NULL</span>,
        fillVar <span class="op">=</span> <span class="st">"sign"</span>,
        fillLab <span class="op">=</span> <span class="st">"Genetic interaction"</span>,
        fillCol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#6D90CA"</span>, <span class="st">"#F6EB13"</span>, <span class="st">"grey80"</span><span class="op">)</span>,
        sizeVar <span class="op">=</span> <span class="st">"fold_enrichment"</span>,
        sizeLab <span class="op">=</span> <span class="st">"Fold enrichment"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>
        panel.grid.major.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"italic"</span>, angle <span class="op">=</span> <span class="fl">90</span>,
        vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></code></pre></div>
<p><img src="figures/fedupDotPlot_sum-1.png" width="100%"></p>
<p>Ok cool, that’s easier to look at than the plot before. Now let’s summarize these pathways even more efficiently using EnrichmentMap!</p>
</div>
<div id="enrichment-map" class="section level2">
<h2 class="hasAnchor">
<a href="#enrichment-map" class="anchor"></a>Enrichment map</h2>
<p>First, make sure to have <a href="https://cytoscape.org/download.html">Cytoscape</a> downloaded and and open on your computer. You’ll also need to install the <a href="http://apps.cytoscape.org/apps/enrichmentmap">EnrichmentMap</a> (≥ v3.3.0) and <a href="http://apps.cytoscape.org/apps/autoannotate">AutoAnnotate</a> apps.</p>
<p>Then format results for compatibility with EnrichmentMap using <code>writeFemap</code>:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">resultsFolder</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="../reference/writeFemap.html">writeFemap</a></span><span class="op">(</span><span class="va">fedupRes</span>, <span class="va">resultsFolder</span><span class="op">)</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_FASN_negative.txt</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_FASN_positive.txt</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_ACACA_negative.txt</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_ACACA_positive.txt</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_LDLR_negative.txt</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_LDLR_positive.txt</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_SREBF1_negative.txt</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_SREBF1_positive.txt</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_SREBF2_negative.txt</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_SREBF2_positive.txt</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_C12orf49_negative.txt</span>
<span class="co">#&gt; Wrote out EM-formatted fedup results file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/femap_C12orf49_positive.txt</span></code></pre></div>
<p>Prepare a pathway annotation file (gmt format) from the pathway list you passed to <code>runFedup</code> using the <code>writePathways</code> function (you don’t need to run this function if your pathway annotations are already in gmt format, but it doesn’t hurt to make sure):</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gmtFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span><span class="op">(</span><span class="st">"pathwaysGMT"</span>, fileext <span class="op">=</span> <span class="st">".gmt"</span><span class="op">)</span>
<span class="fu"><a href="../reference/writePathways.html">writePathways</a></span><span class="op">(</span><span class="va">pathwaysGMT</span>, <span class="va">gmtFile</span><span class="op">)</span>
<span class="co">#&gt; Wrote out pathway gmt file to /var/folders/mh/_0z2r5zj3k75yhtgm6l7xy3m0000gn/T//RtmpEHDYgL/pathwaysGMT85357eb71527.gmt</span></code></pre></div>
<p>Cytoscape is open right? If so, run these lines and let the <code>plotFemap</code> magic happen:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">netFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span><span class="op">(</span><span class="st">"fedupEM_geneMulti"</span>, fileext <span class="op">=</span> <span class="st">".png"</span><span class="op">)</span>
<span class="fu"><a href="../reference/plotFemap.html">plotFemap</a></span><span class="op">(</span>
    gmtFile <span class="op">=</span> <span class="va">gmtFile</span>,
    resultsFolder <span class="op">=</span> <span class="va">resultsFolder</span>,
    qvalue <span class="op">=</span> <span class="fl">0.05</span>,
    chartData <span class="op">=</span> <span class="st">"DATA_SET"</span>,
    hideNodeLabels <span class="op">=</span> <span class="cn">TRUE</span>,
    netName <span class="op">=</span> <span class="st">"fedupEM_geneMulti"</span>,
    netFile <span class="op">=</span> <span class="va">netFile</span>
<span class="op">)</span></code></pre></div>
<p><img src="figures/fedupEM_geneMulti.png"></p>
<p>To note here, the EM nodes were coloured manually (by a similar palette of colours passed to <code>plotDotPlot</code>) in Cytoscape via the <em>Change Colors</em> option in the EM panel. A feature for automated dataset colouring is set to be released in <a href="https://github.com/BaderLab/EnrichmentMapApp/issues/455">version 3.3.2</a> of EnrichmentMap.</p>
<p>This has effectively summarized the 244 pathways from our dot plot into 27 unique biological themes (including 7 unclustered pathways). We can now see clear themes in the data pertaining to negative and positive genetic interactions related to our genes of interest.</p>
</div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.0 (2021-05-18)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Big Sur 10.16</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_CA.UTF-8/en_CA.UTF-8/en_CA.UTF-8/C/en_CA.UTF-8/en_CA.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] ggplot2_3.3.5  tidyr_1.1.3    dplyr_1.0.7    fedup_1.0.0    colorout_1.2-2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] bitops_1.0-7        matrixStats_0.59.0  fs_1.5.0           </span>
<span class="co">#&gt;   [4] usethis_2.0.1       devtools_2.4.2      RColorBrewer_1.1-2 </span>
<span class="co">#&gt;   [7] httr_1.4.2          rprojroot_2.0.2     repr_1.1.3         </span>
<span class="co">#&gt;  [10] tools_4.1.0         backports_1.2.1     bslib_0.2.5.1      </span>
<span class="co">#&gt;  [13] utf8_1.2.1          R6_2.5.0            DBI_1.1.1          </span>
<span class="co">#&gt;  [16] BiocGenerics_0.38.0 colorspace_2.0-2    withr_2.4.2        </span>
<span class="co">#&gt;  [19] uchardet_1.1.0      tidyselect_1.1.1    prettyunits_1.1.1  </span>
<span class="co">#&gt;  [22] processx_3.5.2      curl_4.3.2          compiler_4.1.0     </span>
<span class="co">#&gt;  [25] textshaping_0.3.5   graph_1.70.0        cli_3.0.0          </span>
<span class="co">#&gt;  [28] desc_1.3.0          labeling_0.4.2      sass_0.4.0         </span>
<span class="co">#&gt;  [31] scales_1.1.1        callr_3.7.0         pkgdown_1.6.1      </span>
<span class="co">#&gt;  [34] pbdZMQ_0.3-5        systemfonts_1.0.2   stringr_1.4.0      </span>
<span class="co">#&gt;  [37] digest_0.6.27       rmarkdown_2.9       R.utils_2.10.1     </span>
<span class="co">#&gt;  [40] base64enc_0.1-3     pkgconfig_2.0.3     htmltools_0.5.1.1  </span>
<span class="co">#&gt;  [43] sessioninfo_1.1.1   highr_0.9           fastmap_1.1.0      </span>
<span class="co">#&gt;  [46] ggthemes_4.2.4      rlang_0.4.11        farver_2.1.0       </span>
<span class="co">#&gt;  [49] jquerylib_0.1.4     generics_0.1.0      jsonlite_1.7.2     </span>
<span class="co">#&gt;  [52] zip_2.2.0           R.oo_1.24.0         RCurl_1.98-1.3     </span>
<span class="co">#&gt;  [55] magrittr_2.0.1      Matrix_1.3-4        Rcpp_1.0.6         </span>
<span class="co">#&gt;  [58] IRkernel_1.2        munsell_0.5.0       fansi_0.5.0        </span>
<span class="co">#&gt;  [61] lifecycle_1.0.0     R.methodsS3_1.8.1   stringi_1.6.2      </span>
<span class="co">#&gt;  [64] yaml_2.2.1          MASS_7.3-54         RJSONIO_1.3-1.4    </span>
<span class="co">#&gt;  [67] pkgbuild_1.2.0      plyr_1.8.6          grid_4.1.0         </span>
<span class="co">#&gt;  [70] parallel_4.1.0      forcats_0.5.1       crayon_1.4.1       </span>
<span class="co">#&gt;  [73] lattice_0.20-44     IRdisplay_1.0       knitr_1.33         </span>
<span class="co">#&gt;  [76] ps_1.6.0            pillar_1.6.1        igraph_1.2.6       </span>
<span class="co">#&gt;  [79] uuid_0.1-4          base64url_1.4       stats4_4.1.0       </span>
<span class="co">#&gt;  [82] pkgload_1.2.1       XML_3.99-0.6        glue_1.4.2         </span>
<span class="co">#&gt;  [85] evaluate_0.14       data.table_1.14.0   remotes_2.4.0      </span>
<span class="co">#&gt;  [88] png_0.1-7           vctrs_0.3.8         testthat_3.0.4     </span>
<span class="co">#&gt;  [91] gtable_0.3.0        purrr_0.3.4         assertthat_0.2.1   </span>
<span class="co">#&gt;  [94] cachem_1.0.5        openxlsx_4.2.4      xfun_0.24          </span>
<span class="co">#&gt;  [97] ragg_1.1.3          RCy3_2.12.3         signal_0.7-7       </span>
<span class="co">#&gt; [100] tibble_3.1.2        dplR_1.7.2          memoise_2.0.0      </span>
<span class="co">#&gt; [103] ellipsis_0.3.2</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Catherine Ross.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>




  </body>
</html>
